x/concentrated-liquidity/swaps.go

```diff
@@ -660,39 +671,67 @@ func (k Keeper) updatePoolForSwap(
 	// Fixed gas consumption per swap to prevent spam
 	poolId := pool.GetId()
 	ctx.GasMeter().ConsumeGas(gammtypes.BalancerGasFeeForSwap, "cl pool swap computation")
-	pool, err := k.getPoolById(ctx, poolId)
-	if err != nil {
-		return err
-	}
 
 	// Spread factors should already be rounded up to a whole number dec, but we do this as a precaution
 	spreadFactorsRoundedUp := sdk.NewCoin(swapDetails.TokenIn.Denom, totalSpreadFactors.Ceil().TruncateInt())
 
 	// Remove the spread factors from the input token
 	swapDetails.TokenIn.Amount = swapDetails.TokenIn.Amount.Sub(spreadFactorsRoundedUp.Amount)
 
-	// Send the input token from the user to the pool's primary address
-	err = k.bankKeeper.SendCoins(ctx, swapDetails.Sender, pool.GetAddress(), sdk.Coins{
-		swapDetails.TokenIn,
-	})
-	if err != nil {
-		return types.InsufficientUserBalanceError{Err: err}
-	}
-
 	// Send the spread factors taken from the input token from the user to the pool's spread factor account
 	if !spreadFactorsRoundedUp.IsZero() {
-		err = k.bankKeeper.SendCoins(ctx, swapDetails.Sender, pool.GetSpreadRewardsAddress(), sdk.Coins{
+		err := k.bankKeeper.SendCoins(ctx, swapDetails.Sender, pool.GetSpreadRewardsAddress(), sdk.Coins{
 			spreadFactorsRoundedUp,
 		})
 		if err != nil {
 			return types.InsufficientUserBalanceError{Err: err}
 		}
 	}
 
+	// InputOutputCoins performs multi-send functionality. It accepts a series of
+	// inputs that correspond to a series of outputs. It returns an error if the
+	// inputs and outputs don't lineup or if any single transfer of tokens fails.
+
+	// inputs := []banktypes.Input{
+	// 	{
+	// 		Address: swapDetails.Sender.String(),
+	// 		Coins:   sdk.NewCoins(swapDetails.TokenIn),
+	// 	},
+	// 	{
+	// 		Address: pool.GetAddress().String(),
+	// 		Coins:   sdk.NewCoins(swapDetails.TokenOut),
+	// 	},
+	// }
+
+	// outputs := []banktypes.Output{
+	// 	{
+	// 		Address: pool.GetAddress().String(),
+	// 		Coins:   sdk.NewCoins(swapDetails.TokenIn),
+	// 	},
+	// 	{
+	// 		Address: swapDetails.Sender.String(),
+	// 		Coins:   sdk.NewCoins(swapDetails.TokenOut),
+	// 	},
+	// }
+
+	// err := k.bankKeeper.InputOutputCoins(ctx, inputs, outputs)
+	// if err != nil {
+	// 	return types.InsufficientPoolBalanceError{Err: err}
+	// }
+
+	// Send the input token from the user to the pool's primary address
+	err := k.bankKeeper.SendCoins(ctx, swapDetails.Sender, pool.GetAddress(), sdk.Coins{
```

Comment:
note: i tried using [multisend api](https://github.com/osmosis-labs/osmosis/pull/5850/files#diff-014d8f3d7938e869988af2d7f7ada29412dd1c7490e7cbcd83fa4f3903ef7e09R691-R720) to replace multiple sendCoins call, but the keeper function doesnot take multiple input address 
ref: https://github.com/osmosis-labs/cosmos-sdk/blob/osmosis-main/x/bank/types/msgs.go#L174-L175

Reply:
Right, we will want to still do multiple multi-sends, just only one send per sender!